### ISSUE TITLE
Problem with FeelFMI - wrong results

### ISSUE THREAD

**Issue by lsala** [feelpp/feelpp#1278]
The Feel-FMI library is not working as expected, the numerical results are wrong.

To test it in `testsuite/feelfmi`:
```
./feelpp_test_fmi
```
or simply
```
ctest
```

**Comment by lsala** [comment]
In another test run locally it seems that in the first step of a simulation it reaches already the final solution at time `Tend`that we have specified and it keeps on that without changing
In the case I’m presenting I have chosen a time step of `0.0002`, a initial time of `0.0` and time end of `5`.
If we take `Pi_1.phi` as reference (we can take any other output) I’ve checked with the exact solution and `Pi_1.phi (5) = 16.0996` that is the value that is computed from the first step till the end of the simulation.
Here it is the head of the output file:
```
    1 t,Pi_1.phi,Pi_2.phi,Rout.n.v,Rbuffer.p.v,R.i
    2 0.000000,11.5,12.5,15.5,11.5,-1
    3 0.000200,16.0996,16.7526,18,16.0996,-0.652925
    4 0.000400,16.0996,16.7526,18,16.0996,-0.652925
    5 0.000600,16.0996,16.7526,18,16.0996,-0.652925
    6 0.000800,16.0996,16.7526,18,16.0996,-0.652925
    7 0.001000,16.0996,16.7526,18,16.0996,-0.652925
    8 0.001200,16.0996,16.7526,18,16.0996,-0.652925
    9 0.001400,16.0996,16.7526,18,16.0996,-0.652925
   10 0.001600,16.0996,16.7526,18,16.0996,-0.652925
   11 0.001800,16.0996,16.7526,18,16.0996,-0.652925
   12 0.002000,16.0996,16.7526,18,16.0996,-0.652925
   13 0.002200,16.0996,16.7526,18,16.0996,-0.652925
   14 0.002400,16.0996,16.7526,18,16.0996,-0.652925
   15 0.002600,16.0996,16.7526,18,16.0996,-0.652925
   16 0.002800,16.0996,16.7526,18,16.0996,-0.652925
   17 0.003000,16.0996,16.7526,18,16.0996,-0.652925
   18 0.003200,16.0996,16.7526,18,16.0996,-0.652925
   19 0.003400,16.0996,16.7526,18,16.0996,-0.652925
   20 0.003600,16.0996,16.7526,18,16.0996,-0.652925
   21 0.003800,16.0996,16.7526,18,16.0996,-0.652925
```

**Comment by prudhomm** [comment]
how about om and omc version ? any changes ?

**Comment by prudhomm** [comment]
so values are not being updated as they should after the first time step

**Comment by lsala** [comment]
The test was realized on Atlas in `release` and `debug` mode.
The omc installed is 1.13.0

**Comment by prudhomm** [comment]
is the final value correct or not  ? eg 16.0996

**Comment by lsala** [comment]
Actually I was more thinking that in the first step is already running till the end and then it's not updating since the system has already reached the final time

**Comment by prudhomm** [comment]
could be that too indeed

**Comment by lsala** [comment]
The final error is 10^(-8) so i's the final solution is the correct one

**Comment by prudhomm** [comment]
ok then could it be something changed in OM/OMC ?

**Comment by lsala** [comment]
I've never changed it, or it has been changed on Atlas or I've never changed the settings. 
Have you tested it elsewhere (not Atlas)?

**Comment by prudhomm** [comment]
not yet, I will do asap

**Comment by lsala** [comment]
In the containers and locally using OpenModelica version 1.12 everything is fine.

The problem is using the OpenModelica version installed on Atlas, which is 1.13

**Comment by lsala** [comment]
I made another couple of tests to understand where the problem is and it seems that the wrapper `fmi2_import_do_step` is already giving "wrong" results.
Code line: https://github.com/feelpp/feelpp/blob/develop/feelpp/feel/feelfmi/fmumodel2.cpp#L173
 
This function has as main input the fmu, the current time and the time step.
All these input are correct, however it returns the results at time `t=t_final` instead of computing the results at `t=t_cur+step`

In the singularity image, everything is fine and there are no problem at all.

**Comment by prudhomm** [comment]
@lsala I am confused, the wrong results are obtained on atlas right ? the code is not the culprit but rather the installation of the software, right?

/cc @boileaum

**Comment by lsala** [comment]
I think so since not in atlas is working and the wrapper is defined inside the openmodelica/fmi libraries

**Comment by prudhomm** [comment]
This is now fixed for latest OM version see #1399
